core/services/bulletprooftxmanager/bulletprooftxmanager.go

```diff
@@ -253,42 +259,55 @@ func (b *BulletproofTxManager) Trigger(addr common.Address) {
 	}
 }
 
+type NewTx struct {
+	FromAddress    common.Address
+	ToAddress      common.Address
+	EncodedPayload []byte
+	GasLimit       uint64
+	Meta           *EthTxMeta
+
+	Strategy TxStrategy
+}
+
 // CreateEthTransaction inserts a new transaction
-func (b *BulletproofTxManager) CreateEthTransaction(db *gorm.DB, fromAddress, toAddress common.Address, payload []byte, gasLimit uint64, meta interface{}, strategy TxStrategy) (etx EthTx, err error) {
-	err = CheckEthTxQueueCapacity(db, fromAddress, b.config.EvmMaxQueuedTransactions())
+func (b *BulletproofTxManager) CreateEthTransaction(db *gorm.DB, newTx NewTx) (etx EthTx, err error) {
+	err = CheckEthTxQueueCapacity(db, newTx.FromAddress, b.config.EvmMaxQueuedTransactions())
 	if err != nil {
 		return etx, errors.Wrap(err, "BulletproofTxManager#CreateEthTransaction")
 	}
 
-	// meta can hold arbitrary data and is mostly useful for logging/debugging
-	var metaBytes []byte
-	if meta != nil {
-		metaBytes, err = json.Marshal(meta)
-		if err != nil {
-			return etx, errors.Wrap(err, "BulletproofTxManager#CreateEthTransaction failed to marshal ethtx metadata")
-		}
-	}
-
 	value := 0
 	err = postgres.GormTransactionWithDefaultContext(db, func(tx *gorm.DB) error {
+		if newTx.Meta != nil && newTx.Meta.PipelineTaskRunID != nil {
+			err = tx.Raw(`SELECT * FROM eth_txes WHERE meta->>'PipelineTaskRunID' = ?`, newTx.Meta.PipelineTaskRunID).Scan(&etx).Error
```

Comment:
Can we make `pipeline_task_run_id` and `min_confirmations` normal columns?

It's so interlinked now I think any attempt at encapsulation is going to make our own lives more difficult and introduce potential performance problems.